### PR TITLE
Add documentation about binary format and other internals

### PR DESCRIPTION
**Pull request by klaussfreire**

### CODE REVIEW

File: doc/source/internals/ondisk/objects.rst [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+            uint32 ui32;
+            int64 i64;
+            uint64 ui64;
+            float f;
+            double d;
+            bool b;
+        }
+    }
+
+As can be gleaned from the above, the value itself won't be aligned except for the byte-sized types, and
```
Comment (llogiudice) [inline doc/source/internals/ondisk/objects.rst]: This doesn't match the C code. A `union` has the needed size and alignment to hold any of its constituent members.

File: doc/source/internals/ondisk/sequences.rst [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+
+    union VarLength {
+        // For typecodes l, L, d, t, T
+        long longlength:56;
+
+        // For other typecodes, length < 0xFFFFFF
+        int shortlength:24;
+
+        struct {
+            int padding:24; // 0xFFFFFF
```
Comment (llogiudice) [inline doc/source/internals/ondisk/sequences.rst:18]: What purpose does this `padding` member serve ?

**Comment by llogiudice** [comment]
I'm not really fond of using `long long`, `unsigned long long` et al to refer to fixed-size C-like integers. Since their sizes aren't standardized, it can lead to confusion when using them in specific platforms or toolchains. I'd much rather use the nomenclature defined in `<stdint.h> `, much like numpy does.

File: doc/source/internals/ondisk/sequences.rst (hunk 2 above)
Comment (klaussfreire) [reply doc/source/internals/ondisk/sequences.rst:18]: It helps identify long lengths when reading the buffer. Think of this as rice coding.

File: doc/source/CHANGELOG.rst [hunk 3]
```diff
@@ -5,7 +5,14 @@ All notable changes to this project will be documented here.
 
 The format is largely inspired by keepachangelog_.
 
-.. _0.1.1:
+v0.7.1 - Unreleased
```
Comment (tzulberti-jampp) [inline doc/source/CHANGELOG.rst]: I don't think this is a bugfix

File: doc/source/internals/ondisk/datetime.rst [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    ts = (int(timestamp) << 20) + frac(timestamp) * 1000000
+
+Examples
+--------
+
+.. code-block:: pycon
+
+    >>> buf = bytearray(1 << 20)
+    >>> end = mapped_datetime.pack_into(datetime.datetime(2019, 03, 05, 11, 15, 24, 388001), buf, 0)
```
Comment (tzulberti-jampp) [inline doc/source/internals/ondisk/datetime.rst:27]: Check the `03` and `05`. They are octal values. Is that required?

File: doc/source/CHANGELOG.rst (hunk 3 above)
Comment (klaussfreire) [reply doc/source/CHANGELOG.rst]: It isn't. But it's not going to be released on its own either. The final version will be assigned whenever something release-worthy gets pushed. This is more like a pre-emptive version bump.

Comment (klaussfreire) [reply doc/source/CHANGELOG.rst]: In fact, I think this is conflicting with master by now, and by now master is on 0.8.0, illustrating the point.

File: doc/source/internals/ondisk/datetime.rst (hunk 4 above)
Comment (klaussfreire) [reply doc/source/internals/ondisk/datetime.rst:27]: It's just an exaple. It's probably not necessary, but it was pasted from a console, so I'm sure it works.

**Comment by klaussfreire** [comment]
Ping @tzulberti-jampp @llogiudice and friends

**Pull request merged**